zp-relayer/workers/sentTxWorker.ts

```diff
@@ -99,55 +124,81 @@ export async function createSentTxWorker<T extends EstimationType>(gasPrice: Gas
           logger.error('Commitments are not equal')
         }
 
-        return txHash
+        return [SentTxState.MINED, txHash] as SentTxResult
       } else {
         // Revert
         logger.error('%s Transaction %s reverted at block %s', logPrefix, txHash, tx.blockNumber)
 
-        // TODO: a more efficient strategy would be to collect all other jobs
-        // and move them to 'failed' state as we know they will be reverted
-        // To do this we need to acquire a lock for each job. Did not find
-        // an easy way to do that yet. See 'collectBatch'
-        const jobs = await sentTxQueue.getJobs(['delayed', 'waiting'])
-        const ids = jobs.map(j => j.id as string)
-        logger.info('%s marking ids %j as failed', logPrefix, ids)
-        await markFailed(ids)
-
-        logger.info('Rollback optimistic state...')
-        pool.optimisticState.rollbackTo(pool.state)
-        logger.info('Clearing optimistic nullifiers...')
-        await pool.optimisticState.nullifiers.clear()
-        logger.info('Clearing optimistic roots...')
-        await pool.optimisticState.roots.clear()
-
-        const root1 = pool.state.getMerkleRoot()
-        const root2 = pool.optimisticState.getMerkleRoot()
-        logger.info(`Assert roots are equal: ${root1}, ${root2}, ${root1 === root2}`)
+        await clearOptimisticState()
+        return [SentTxState.REVERT, txHash] as SentTxResult
       }
     } else {
+      // Resend with updated gas price
+      if (txType === TxType.PERMITTABLE_DEPOSIT) {
+        const deadline = (txData as PermittableDepositTxData).deadline
+        await checkAssertion(() => checkDeadline(deadline, config.permitDeadlineThresholdResend))
```

Comment:
Can you clarify, what will happen with the this particular job if the deadline will be expired at this point? What will happen with the jobs that are positioned later in the optimistic state?

Reply:
This part with deadline check should be moved inside try-catch block below (just before transaction submission), so that this error will be caught and state rollback is performed. Will fix now.